user_profile/urls.py

```diff
@@ -15,4 +15,5 @@
     path("profile/", views.view_profile, name="view_profile"),
     path("profile/edit", views.edit_profile, name="edit_profile"),
     path("profile/delete", views.milestone_profile, name="milestone_profile"),
+    path("profile/detail/<int:id>", views.detail_profile, name="detail_profile"),
```

Comment:
where is this id coming from

Reply:
This would be coming from the user model.

id is the [primary key](https://stackoverflow.com/questions/22345711/whats-the-difference-between-model-id-and-model-pk-in-django#:~:text=pk%20is%20the%20attribute%20that,if%20none%20is%20explicitly%20specified.) of any model where the primary key is not explicitly specified.